python/cudf/dataframe/index.py

```diff
@@ -341,8 +341,10 @@ def copy(self, deep=True):
         return result
 
     def __repr__(self):
-        return "{}(start={}, stop={})".format(self.__class__.__name__,
-                                              self._start, self._stop)
+        return "{}(start={}, stop={}".format(self.__class__.__name__,
+                                             self._start, self._stop)\
+                + (", name='{}'".format(str(self.name))
+                   if self.name is not None else "") + ")"
```

Comment:
We are presently not printing name of certain types of index's. So fixed that as well.